File: Dataset/Leetcode/valid/13/40.java
class Solution {
    public int XXX(String s) {
        int status = 0;
        int result = 0;
        for (int i = 0; i < s.length(); i++) {
            switch (s.charAt(i)) {
                case 'I':
                    result += 1;
                    status = 1;
                    break;
                case 'V':
                    result += 5;
                    if (status == 1) {
                        result -= 2;
                    }
                    break;
                case 'X':
                    result += 10;
                    if (status == 1) {
                        result -= 2;
                    }
                    status = 2;
                    break;
                case 'L':
                    result += 50;
                    if (status == 2) {
                        result -= 20;
                    }
                    break;
                case 'C':
                    result += 100;
                    if (status == 2) {
                        result -= 20;
                    }
                    status = 3;
                    break;
                case 'D':
                    result += 500;
                    if (status == 3) {
                        result -= 200;
                    }
                    break;
                case 'M':
                    result += 1000;
                    if (status == 3) {
                        result -= 200;
                    }
                    break;
                default:
            }

        }
        return result;
    }
}

File: Dataset/Leetcode/train/125/40.cpp
 class Solution {
public:
    bool XXX(string s) {
        int i=0, j=s.size()-1;
        while(i < j){
            while(i < j && tran(s[i]) == -1) i ++;
            while(i < j && tran(s[j]) == -1) j --;
            if(i < j){
                if(tran(s[i]) != tran(s[j])) return false;
                i ++; j--;
            }
        }
        return true;
    }
    
    int tran(char x){
        if((x>='a' && x<='z') || (x>='0' && x<='9')) return int(x);
        else if(x>='A' && x<='Z') return int(x+32);
        return -1;
    }
};

